### ISSUE TITLE
🐛 Debian 11 dpkg package doesn’t support zstd compression.

### ISSUE THREAD

**Issue by DarkPark** [dandavison/delta#1248]
```sh
sudo dpkg -i /home/user/software/git-delta_0.15.0_amd64.deb 
dpkg-deb: error: archive '/home/user/software/git-delta_0.15.0_amd64.deb' uses unknown compression for member 'control.tar.zst', giving up
dpkg: error processing archive /home/user/software/git-delta_0.15.0_amd64.deb (--install):
 dpkg-deb --control subprocess returned error exit status 2
Errors were encountered while processing:
 /home/user/software/git-delta_0.15.0_amd64.deb
```

**Comment by dandavison** [comment]
Thanks @DarkPark. I don't have any insight into this currently. The changelog entry for this release is https://github.com/dandavison/delta/releases/tag/0.15.0 . I don't immediately see anything there that would cause this. I do see recent google hits for other people encountering this error message (`"uses unknown compression for member 'control.tar.zst'"`). The delta codebase doesn't contain the string "zst".

Random uninformed guess: Is it to do with the .debs being built on an Ubuntu machine rather than a Debian machine by Github Actions?

Any ideas @baryluk @MarcoIeni ?

**Comment by MarcoIeni** [comment]
This might be due to a recent change in github actions, where ubuntu-latest has become ubuntu 22 instead of ubuntu 20.

**Comment by baryluk** [comment]
I guess it is a default on newer machine.

On my Debian testing machine, in manual of `dpkg-buildpackage` and `dpkg-source`  there are options to switch compression type and level, but I do not see `zstd` there, only `gzip`, `bzip2`, `lzma` and `xz`. And it default to `xz`.

My guess is indeed that this is Ubuntu specific extension, that was not merged into Debian (having it merged would also meen that now libzstd is an essential package, which would need to be installed on every Debian system, which is not something to take lightly). While `zstd` has faster compression and decompression, it is not as efficient as `xz`, thus using more space on mirror and using more network bandwidth (multiply that by 50000 packages and millions of users). I would suggest passing explicitly `--compression=xz` to `dpkg-buildpackage`.

**Comment by baryluk** [comment]
I can confirm that the released deb package also does not install on my Debian unstable system with the same error.

I checked dpkg (including `dpkg-dev`) git repo https://git.dpkg.org/cgit/dpkg/dpkg.git , and I could not find any mention of zstd.

I do see it in the latest Ubuntu fork of this repo: https://git.launchpad.net/ubuntu/+source/dpkg/commit/?id=c12d3702f1deef466ee7840c81ea1cbd5557f9fd

**Comment by dandavison** [comment]
I don't have a debian/ubuntu machine to hand, but I would be happy to issue a second release today to fix this problem.

> I would suggest passing explicitly `--compression=xz` to `dpkg-buildpackage`.

Would it be this line we'd modify?

https://github.com/dandavison/delta/blob/b4ae4c47fbd82fbdcb04271daa7c34b801814fe6/etc/ci/before_deploy.sh#L165

I see https://man7.org/linux/man-pages/man1/dpkg-deb.1.html

```
       -Zcompress-type
              Specify which compression type to use when building a
              package.  Allowed values are gzip, xz (since dpkg 1.15.6),
              and none (default is xz).
```

Does anyone with an Ubuntu/Debian machine feel like making a PF for this??

**Comment by baryluk** [comment]
> Would it be this line we'd modify?

@dandavison Yes. You can pass `-Zxz` to `dpkg-deb` (can also be set via env variable `DPKG_DEB_COMPRESSOR_TYPE=xz`)

**Comment by dandavison** [comment]
I've made that change and have made a new release https://github.com/dandavison/delta/releases/tag/0.15.1 (@baryluk mentioned that they'd tested the `dpkg-deb` command, so that's good enough to try a new release seeing as it's a simple change.)

**Comment by baryluk** [comment]
Looks good!

```shell
root@debian:/home/user# dpkg -i git-delta_0.15.1_amd64.deb 
Selecting previously unselected package git-delta.
(Reading database ... 1091213 files and directories currently installed.)
Preparing to unpack git-delta_0.15.1_amd64.deb ...
Unpacking git-delta (0.15.1) ...
Setting up git-delta (0.15.1) ...
root@debian:/home/user# 
```

```shell
user@debian:~$ dpkg -L git-delta
/.
/usr
/usr/bin
/usr/bin/delta
/usr/share
/usr/share/doc
/usr/share/doc/git-delta
/usr/share/doc/git-delta/README.md
/usr/share/doc/git-delta/copyright
user@debian:~$ delta --version
delta 0.15.1
user@debian:~$ file `which delta`
/usr/bin/delta: ELF 64-bit LSB pie executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/ld-linux-x86-64.so.2, BuildID[sha1]=207edd16d2ad1ab3996999d77c7833d522e5cf6c, for GNU/Linux 3.2.0, stripped
```

**Comment by DarkPark** [comment]
Unfortunately now there is a new problem during installation.
Debian 11 stable.

```bash
$ sudo dpkg -i /home/stas/software/git-delta_0.15.1_amd64.deb 
(Reading database ... 197575 files and directories currently installed.)
Preparing to unpack .../git-delta_0.15.1_amd64.deb ...
Unpacking git-delta (0.15.1) over (0.14.0) ...
dpkg: dependency problems prevent configuration of git-delta:
 git-delta depends on libc6 (>= 2.34); however:
  Version of libc6:amd64 on system is 2.31-13+deb11u5.

dpkg: error processing package git-delta (--install):
 dependency problems - leaving unconfigured
Errors were encountered while processing:
 git-delta
```

**Comment by baryluk** [comment]
@DarkPark Probably better to open a new issue for this. It is just built with two new glibc.

![delta-deps](https://user-images.githubusercontent.com/110322/205462578-92e6d07e-22ab-4c9c-ab4d-c90ca04b4ce3.png)

The only easy and reliable solution to this problem is to build using older glibc. (There are other methods, but they are not easy).

**Comment by DarkPark** [comment]
@baryluk ok, will do.
